src/Backend/ONE/ONEToolchain.ts

```diff
@@ -14,33 +14,106 @@
  * limitations under the License.
  */
 
+import {execSync} from 'child_process';
 import {Backend} from '../Backend';
 import {Command} from '../Command';
 import {Compiler} from '../Compiler';
 import {Executor} from '../Executor';
-import {Toolchains} from '../Toolchain';
+import {ToolchainInfo, Toolchains} from '../Toolchain';
+import {DebianArch, DebianRepo, DebianToolchain} from '../ToolchainImpl/DebianToolchain';
+import {Version} from '../Version';
 
 class ToolchainCompiler implements Compiler {
   private readonly toolchainTypes: string[];
+  private readonly toolchainName: string;
+  private readonly debianRepo: DebianRepo;
+  private readonly debianArch: DebianArch;
 
   constructor() {
-    this.toolchainTypes = [];
+    this.toolchainName = 'onecc-docker-test';
+    // we currently support only the latest version of one-compiler.
+    this.toolchainTypes = ['latest'];
+
+    // we temporary add personal PPA for test.
+    this.debianRepo =
+        new DebianRepo('http://ppa.launchpad.net/psalmist-kim/ppa/ubuntu', 'bionic', 'main');
+    this.debianArch = DebianArch.amd64;
   }
 
   getToolchainTypes(): string[] {
     return this.toolchainTypes;
   }
 
-  getToolchains(_toolchainType: string, _start: number, _count: number): Toolchains {
-    throw Error('Invalid getToolchains call');
+  getToolchains(_toolchainType: string, _start?: number, _count?: number): Toolchains {
+    if (parseInt(execSync(`apt-cache show ${this.toolchainName} > /dev/null 2>&1; echo $?`)
+                     .toString()
+                     .trim()) !== 0) {
+      throw Error('the prerequisites are not met');
+    }
+
+    const availableToolchains = new Toolchains();
+
+    if (_toolchainType === 'latest') {
+      const versionList =
+          execSync(`apt-cache show ${this.toolchainName} | grep Version | awk '{print $2}'`)
+              .toString()
+              .trim()
+              .split('\n');
+
+      for (const version of versionList) {
+        const [major, minor, patch, option] = version.split(/[.~]+/);
+
+        const toolchainInfo = new ToolchainInfo(
+            this.toolchainName, 'Description: test',
+            new Version(parseInt(major), parseInt(minor), parseInt(patch), option));
```

Comment:
(Suggestion, but not for this PR. :))

The dependencies will be shown in a tooltip of toolchain item when you mouse over the toolchain item in Toolchain View.

The `onecc-docker` does not have explicit dependency.
However, I think it would be good to show which `one-compiler` version this toolchain is dependent on. :)

Reply:
![0926_vscode_cap1](https://user-images.githubusercontent.com/7223627/192186028-5e78059b-535b-4d77-a237-36d47524618c.png)

Reply:
> However, I think it would be good to show which `one-compiler` version this toolchain is dependent on. :)

Thank you for your review! I assume, this suggestion has something to do with Team AndroMETA. Is this right?

Reply:
Oh, this changes is up to us. :) The content that appears here can be solved by you adding the depends information here.

Reply:
After we implement our own DebianToolchain in ONEToolchain, we're going to discuss about it again :)

Reply:
<img width="218" alt="image" src="https://user-images.githubusercontent.com/67946956/192920195-54fade3a-17ca-4f2d-ad3a-5a78510d4f02.png">